app/contribution-hub/ConstraintDetailsModal.tsx

```diff
@@ -0,0 +1,273 @@
+      constraint.title,
+      isNotSatisfy,
+      requirementParamsList,
+      constraintFile
+    );
+  };
+
+  const handleSelectNotSatisfy = (isSatisfy: boolean) => {
+    setIsNotSatisfy(isSatisfy);
+  };
+
+  return (
+    <div className="flex flex-col gap-2 mt-5 relative">
+      <div
+        className="absolute -top-14 cursor-pointer z-[999]"
+        onClick={handleBackToConstraintListModal}
+      >
+        <Icon
+          iconSrc="/assets/images/provider-dashboard/arrow-left.svg"
+          className="cursor-pointer z-[999999]"
+        />
+      </div>
+      <div className="w-full flex gap-4 h-[32px] mb-2">
+        <div
+          onClick={() => handleSelectNotSatisfy(false)}
+          className={`w-full flex items-center justify-center rounded-lg h-full cursor-pointer text-white relative overflow-hidden`}
+        >
+          <div
+            className={`${
+              !isNotSatisfy ? "bg-dark-space-green opacity-30" : "bg-gray50"
+            } absolute w-full h-full`}
+          ></div>
+          <p className="absolute text-white">Should satisfy</p>
+        </div>
+        <div
+          onClick={() => handleSelectNotSatisfy(true)}
+          className={`w-full flex items-center justify-center rounded-lg h-full cursor-pointer text-white relative overflow-hidden`}
+        >
+          <div
+            className={`${
+              isNotSatisfy ? "bg-error opacity-50" : "bg-gray50"
+            } absolute w-full h-full `}
+          ></div>
+          <p className="absolute text-white">Should not satisfy</p>
+        </div>
+      </div>
+      <CreateParams
+        constraint={constraint}
+        setRequirementParamsList={setRequirementParamsList}
+        requirementParamsList={requirementParamsList}
+        constraintFile={constraintFile}
+        setConstraintFile={setConstraintFile}
+        allChainList={allChainList}
+      />
+      <div className="mb-4">{constraint.description}</div>
+      <div className="text-error text-[10px] min-h-[15px]">{errorMessage}</div>
+      <div
+        onClick={handleAddRequirement}
+        className="flex cursor-pointer  bg-gray40 text-[14px] font-semibold text-white h-[44px] border-2 border-gray70 rounded-xl items-center justify-center mb-2"
+      >
+        Add Requirement
+      </div>
+    </div>
+  );
+};
+
+const CreateParams = ({
+  constraint,
+  setRequirementParamsList,
+  requirementParamsList,
+  constraintFile,
+  setConstraintFile,
+  allChainList,
+}: CreateModalParam) => {
+  const [reqNftAddress, setReqNftAddress] = useState("");
+
+  useEffect(() => {
+    if (!requirementParamsList) return;
+    setReqNftAddress(requirementParamsList.COLLECTION_ADDRESS);
+  }, [requirementParamsList]);
+
+  const handleChangeCollection = (address: string) => {
+    setReqNftAddress(address);
+    setRequirementParamsList({
+      ...requirementParamsList,
+      ["COLLECTION_ADDRESS"]: address,
+    });
+  };
+
+  if (constraint.params.length === 0) return;
```

Comment:
do not return `undefined`. return `null` instead

Reply:
fixed